File: crates/tauri-store-cli/assets/commands.ts
// @ts-nocheck
import { commands } from '@tauri-store/shared';

/**
 * Stops the autosave.
 *
 * @example
 * ```ts
 * import { clearAutosave } from 'KEBAB_PLUGIN_FULL_NAME';
 *
 * await clearAutosave();
 * ```
 */
export const clearAutosave = commands.clearAutosave('PLUGIN_NAME');

/**
 * Gets the default save strategy for the stores.
 * It can be overridden on a per-store basis.
 *
 * @example
 * ```ts
 * import { getDefaultSaveStrategy } from 'KEBAB_PLUGIN_FULL_NAME';
 *
 * const { strategy, interval } = await getDefaultSaveStrategy();
 * console.log(strategy, interval);
 * ```
 */
export const getDefaultSaveStrategy = commands.getDefaultSaveStrategy('PLUGIN_NAME');

/**
 * Directory where the stores are saved.
 *
 * @example
 * ```ts
 * import { getPASCAL_PLUGIN_NAMEPath } from 'KEBAB_PLUGIN_FULL_NAME';
 *
 * const path = await getPASCAL_PLUGIN_NAMEPath();
 * console.log(path);
 * ```
 */
export const getPASCAL_PLUGIN_NAMEPath = commands.getStoreCollectionPath('PLUGIN_NAME');

/**
 * Gets the save strategy used by a store.
 *
 * @example
 * ```ts
 * import { getSaveStrategy } from 'KEBAB_PLUGIN_FULL_NAME';
 *
 * const { strategy, interval } = await getSaveStrategy('my-store');
 * console.log(strategy, interval);
 * ```
 **/
export const getSaveStrategy = commands.getSaveStrategy('PLUGIN_NAME');

/**
 * Lists all the store ids.
 *
 * @example
 * ```ts
 * import { getStoreIds } from 'KEBAB_PLUGIN_FULL_NAME';
 *
 * const ids = await getStoreIds();
 * for (const id of ids) {
 *   console.log(id);
 * }
 */
export const getStoreIds = commands.getStoreIds('PLUGIN_NAME');

/**
 * Path where the store is saved.
 *
 * @example
 * ```ts
 * import { getStorePath } from 'KEBAB_PLUGIN_FULL_NAME';
 *
 * const path = await getStorePath('my-store');
 * console.log(path);
 * ```
 **/
export const getStorePath = commands.getStorePath('PLUGIN_NAME');

/**
 * Gets the state of a store.
 *
 * @example
 * ```ts
 * import { getStoreState } from 'KEBAB_PLUGIN_FULL_NAME';
 *
 * const state = await getStoreState('counter-store');
 * console.log(state.counter);
 * ```
 */
export const getStoreState = commands.getStoreState('PLUGIN_NAME');

/**
 * Saves a store to the disk.
 *
 * @example
 * ```ts
 * import { save } from 'KEBAB_PLUGIN_FULL_NAME';
 *
 * // Save a single store.
 * await save('my-store');
 *
 * // Save some stores.
 * await save('my-store', 'my-store-2');
 * ```
 **/
export const save = commands.save('PLUGIN_NAME');

/**
 * Saves all the stores to the disk.
 *
 * @example
 * ```ts
 * import { saveAll } from 'KEBAB_PLUGIN_FULL_NAME';
 *
 * await saveAll();
 * ```
 */
export const saveAll = commands.saveAll('PLUGIN_NAME');

/**
 * Saves all the stores to the disk immediately, ignoring the save strategy.
 *
 * @example
 * ```ts
 * import { saveAllNow } from 'KEBAB_PLUGIN_FULL_NAME';
 *
 * await saveAllNow();
 * ```
 */
export const saveAllNow = commands.saveAllNow('PLUGIN_NAME');

/**
 * Saves a store to the disk immediately, ignoring the save strategy.
 *
 * @example
 * ```ts
 * import { saveNow } from 'KEBAB_PLUGIN_FULL_NAME';
 *
 * // Save a single store.
 * await saveNow('my-store');
 *
 * // Save some stores.
 * await saveNow('my-store', 'my-store-2');
 * ```
 */
export const saveNow = commands.saveNow('PLUGIN_NAME');

/**
 * Saves the stores periodically.
 * @param interval The interval in milliseconds.
 *
 * @example
 * ```ts
 * import { setAutosave } from 'KEBAB_PLUGIN_FULL_NAME';
 *
 * // Save all the stores every 5 minutes.
 * await setAutosave(5 * 60 * 1000);
 * ```
 */
export const setAutosave = commands.setAutosave('PLUGIN_NAME');

/**
 * Sets the save strategy for a store.
 * Calling this will abort any pending save operation.
 *
 * @example
 * ```ts
 * import { setSaveStrategy } from 'KEBAB_PLUGIN_FULL_NAME';
 *
 * await setSaveStrategy('my-store', 'debounce', 1000);
 * ```
 */
export const setSaveStrategy = commands.setSaveStrategy('PLUGIN_NAME');

/**
 * Sets the store options.
 *
 * @example
 * ```ts
 * import { setStoreOptions } from 'KEBAB_PLUGIN_FULL_NAME';
 *
 * await setStoreOptions('my-store', {
 *  saveOnChange: true,
 *  saveStrategy: 'debounce',
 *  saveInterval: 1000
 * });
 * ```
 */
export const setStoreOptions = commands.setStoreOptions('PLUGIN_NAME');

/////////////////////////
// INTERNAL COMMANDS
/////////////////////////

/** @internal */
export const load = commands.load('PLUGIN_NAME');

/** @internal */
export const patch = commands.patch('PLUGIN_NAME');

/** @internal */
export const unload = commands.unload('PLUGIN_NAME');

File: crates/tauri-store-cli/assets/commands.rs
use crate::manager::ManagerExt;
use std::path::PathBuf;
use std::time::Duration;
use tauri::{AppHandle, Manager, Runtime, WebviewWindow};
use tauri_store::{with_store, Result, SaveStrategy, StoreOptions, StoreState};

#[cfg(feature = "unstable-async")]
use tauri_store::boxed;

#[tauri::command]
pub(crate) async fn clear_autosave<R: Runtime>(app: AppHandle<R>) {
  app.PLUGIN_NAME().clear_autosave();
}

#[tauri::command]
pub(crate) async fn get_default_save_strategy<R>(app: AppHandle<R>) -> SaveStrategy
where
  R: Runtime,
{
  app.PLUGIN_NAME().default_save_strategy()
}

#[tauri::command]
pub(crate) async fn get_PLUGIN_NAME_path<R: Runtime>(app: AppHandle<R>) -> PathBuf {
  app.PLUGIN_NAME().path().to_path_buf()
}

#[tauri::command]
pub(crate) async fn get_store_ids<R: Runtime>(app: AppHandle<R>) -> Vec<String> {
  app.PLUGIN_NAME().ids()
}

#[cfg(not(feature = "unstable-async"))]
#[tauri::command]
pub(crate) async fn get_store_path<R>(app: AppHandle<R>, id: String) -> Result<PathBuf>
where
  R: Runtime,
{
  with_store(&app, id, |store| store.path())
}

#[cfg(feature = "unstable-async")]
#[tauri::command]
pub(crate) async fn get_store_path<R>(app: AppHandle<R>, id: String) -> Result<PathBuf>
where
  R: Runtime,
{
  with_store(&app, id, |store| boxed(store.path())).await
}

#[cfg(not(feature = "unstable-async"))]
#[tauri::command]
pub(crate) async fn get_save_strategy<R>(app: AppHandle<R>, id: String) -> Result<SaveStrategy>
where
  R: Runtime,
{
  with_store(&app, id, |store| store.save_strategy())
}

#[cfg(feature = "unstable-async")]
#[tauri::command]
pub(crate) async fn get_save_strategy<R>(app: AppHandle<R>, id: String) -> Result<SaveStrategy>
where
  R: Runtime,
{
  with_store(&app, id, |store| boxed(store.save_strategy())).await
}

#[cfg(not(feature = "unstable-async"))]
#[tauri::command]
pub(crate) async fn get_store_state<R>(app: AppHandle<R>, id: String) -> Result<StoreState>
where
  R: Runtime,
{
  app.PLUGIN_NAME().store_state(id)
}

#[cfg(feature = "unstable-async")]
#[tauri::command]
pub(crate) async fn get_store_state<R>(app: AppHandle<R>, id: String) -> Result<StoreState>
where
  R: Runtime,
{
  app.PLUGIN_NAME().store_state(id).await
}

#[cfg(not(feature = "unstable-async"))]
#[tauri::command]
pub(crate) async fn load<R: Runtime>(app: AppHandle<R>, id: String) -> Result<StoreState> {
  with_store(&app, id, |store| store.state().clone())
}

#[cfg(feature = "unstable-async")]
#[tauri::command]
pub(crate) async fn load<R: Runtime>(app: AppHandle<R>, id: String) -> Result<StoreState> {
  with_store(&app, id, |store| boxed(store.state().clone())).await
}

#[cfg(not(feature = "unstable-async"))]
#[tauri::command]
pub(crate) async fn patch<R>(window: WebviewWindow<R>, id: String, state: StoreState) -> Result<()>
where
  R: Runtime,
{
  let app = window.app_handle();
  let label = window.label().to_owned();
  with_store(app, id, move |store| store.patch_with_source(state, label))?
}

#[cfg(feature = "unstable-async")]
#[tauri::command]
pub(crate) async fn patch<R>(window: WebviewWindow<R>, id: String, state: StoreState) -> Result<()>
where
  R: Runtime,
{
  let app = window.app_handle();
  let label = window.label().to_owned();
  with_store(app, id, move |store| {
    Box::pin(async { store.patch_with_source(state, label).await })
  })
  .await?
}

#[cfg(not(feature = "unstable-async"))]
#[tauri::command]
pub(crate) async fn save<R: Runtime>(app: AppHandle<R>, id: String) -> Result<()> {
  app.PLUGIN_NAME().save(id)
}

#[cfg(feature = "unstable-async")]
#[tauri::command]
pub(crate) async fn save<R: Runtime>(app: AppHandle<R>, id: String) -> Result<()> {
  app.PLUGIN_NAME().save(id).await
}

#[cfg(not(feature = "unstable-async"))]
#[tauri::command]
pub(crate) async fn save_all<R: Runtime>(app: AppHandle<R>) -> Result<()> {
  app.PLUGIN_NAME().save_all()
}

#[cfg(feature = "unstable-async")]
#[tauri::command]
pub(crate) async fn save_all<R: Runtime>(app: AppHandle<R>) -> Result<()> {
  app.PLUGIN_NAME().save_all().await
}

#[cfg(not(feature = "unstable-async"))]
#[tauri::command]
pub(crate) async fn save_all_now<R: Runtime>(app: AppHandle<R>) -> Result<()> {
  app.PLUGIN_NAME().save_all_now()
}

#[cfg(feature = "unstable-async")]
#[tauri::command]
pub(crate) async fn save_all_now<R: Runtime>(app: AppHandle<R>) -> Result<()> {
  app.PLUGIN_NAME().save_all_now().await
}

#[cfg(not(feature = "unstable-async"))]
#[tauri::command]
pub(crate) async fn save_now<R: Runtime>(app: AppHandle<R>, id: String) -> Result<()> {
  app.PLUGIN_NAME().save_now(id)
}

#[cfg(feature = "unstable-async")]
#[tauri::command]
pub(crate) async fn save_now<R: Runtime>(app: AppHandle<R>, id: String) -> Result<()> {
  app.PLUGIN_NAME().save_now(id).await
}

#[cfg(not(feature = "unstable-async"))]
#[tauri::command]
pub(crate) async fn save_some<R: Runtime>(app: AppHandle<R>, ids: Vec<String>) -> Result<()> {
  app.PLUGIN_NAME().save_some(&ids)
}

#[cfg(feature = "unstable-async")]
#[tauri::command]
pub(crate) async fn save_some<R: Runtime>(app: AppHandle<R>, ids: Vec<String>) -> Result<()> {
  app.PLUGIN_NAME().save_some(&ids).await
}

#[cfg(not(feature = "unstable-async"))]
#[tauri::command]
pub(crate) async fn save_some_now<R: Runtime>(app: AppHandle<R>, ids: Vec<String>) -> Result<()> {
  app.PLUGIN_NAME().save_some_now(&ids)
}

#[cfg(feature = "unstable-async")]
#[tauri::command]
pub(crate) async fn save_some_now<R: Runtime>(app: AppHandle<R>, ids: Vec<String>) -> Result<()> {
  app.PLUGIN_NAME().save_some_now(&ids).await
}

#[tauri::command]
pub(crate) async fn set_autosave<R: Runtime>(app: AppHandle<R>, interval: u64) {
  app
    .PLUGIN_NAME()
    .set_autosave(Duration::from_millis(interval));
}

#[cfg(not(feature = "unstable-async"))]
#[tauri::command]
pub(crate) async fn set_save_strategy<R>(
  app: AppHandle<R>,
  id: String,
  strategy: SaveStrategy,
) -> Result<()>
where
  R: Runtime,
{
  with_store(&app, id, |store| store.set_save_strategy(strategy))
}

#[cfg(feature = "unstable-async")]
#[tauri::command]
pub(crate) async fn set_save_strategy<R>(
  app: AppHandle<R>,
  id: String,
  strategy: SaveStrategy,
) -> Result<()>
where
  R: Runtime,
{
  with_store(&app, id, move |store| {
    #[expect(clippy::unit_arg)]
    boxed(store.set_save_strategy(strategy))
  })
  .await
}

#[cfg(not(feature = "unstable-async"))]
#[tauri::command]
pub(crate) async fn set_store_options<R>(
  window: WebviewWindow<R>,
  id: String,
  options: StoreOptions,
) -> Result<()>
where
  R: Runtime,
{
  let app = window.app_handle();
  let label = window.label().to_owned();
  with_store(app, id, move |store| {
    store.set_options_with_source(options, label)
  })?
}

#[cfg(feature = "unstable-async")]
#[tauri::command]
pub(crate) async fn set_store_options<R>(
  window: WebviewWindow<R>,
  id: String,
  options: StoreOptions,
) -> Result<()>
where
  R: Runtime,
{
  let app = window.app_handle();
  let label = window.label().to_owned();
  with_store(app, id, move |store| {
    boxed(store.set_options_with_source(options, label))
  })
  .await?
}

#[cfg(not(feature = "unstable-async"))]
#[tauri::command]
pub(crate) async fn unload<R: Runtime>(app: AppHandle<R>, id: String) -> Result<()> {
  app.PLUGIN_NAME().unload_store(&id)
}

#[cfg(feature = "unstable-async")]
#[tauri::command]
pub(crate) async fn unload<R: Runtime>(app: AppHandle<R>, id: String) -> Result<()> {
  app.PLUGIN_NAME().unload_store(&id).await
}